### ISSUE TITLE
Allow creating nullable enum type

### ISSUE THREAD

**Issue by abogushevsky** [denis-zhdanov/kotlin-baker#20]
Some data classes might have constructor parameters with nullable enum types. Currently Kotlin Baker can't create instances of such classes.

**Comment by denis-zhdanov** [comment]
Checked against the latest sources - it works (see an example below).

Please provide more info on how to reproduce the problem.

```
    @Test
    fun `when nullable enum property is unavailable then null is used`() {
        data class Inner(val data: DayOfWeek?)
        data class Outer(val data: Inner)

        val actual = doCreate(Outer::class, emptyMap())
        assertThat(actual).isEqualTo(Outer(Inner(null)))
    }
```

**Comment by abogushevsky** [comment]
Problem is reproduced only if we have Inner type with several properties and one of them is nullable enum and we have Outer type that has collection of Inner. I've added test case for it:

`data class InnerWithOptionalEnum(val testInt: Int, val dayOfWeek: DayOfWeek?)
  data class ObjectsWithOptionalEnumParameterHolder(val data: List<InnerWithOptionalEnum>)

@Test
    fun `when Inner type contains nullable enum parameter and it's not the only property and outer type contains collection of Inner then it's correctly picked up`() {
        val actual = doCreate(
            ObjectsWithOptionalEnumParameterHolder::class,
            mapOf(
                "data[0].testInt" to "1",
                "data[0].dayOfWeek" to "FRIDAY",
                "data[1].testInt" to "2"))
        assertThat(actual).isEqualTo(
            ObjectsWithOptionalEnumParameterHolder(
                listOf(
                    InnerWithOptionalEnum(1, DayOfWeek.FRIDAY),
                    InnerWithOptionalEnum(2, null))))
    }
`

**Comment by abogushevsky** [comment]
Problem is reproduced only if we have Inner type with several properties and one of them is nullable enum and we have Outer type that has collection of Inner. I've added test case for it:

```
data class InnerWithOptionalEnum(val testInt: Int, val dayOfWeek: DayOfWeek?)
data class ObjectsWithOptionalEnumParameterHolder(val data: List)

@Test
fun `when Inner type contains nullable enum parameter and it's not the only property and outer type contains collection of Inner then it's correctly picked up`() {
    val actual = doCreate(
        ObjectsWithOptionalEnumParameterHolder::class,
        mapOf(
            "data[0].testInt" to "1",
            "data[0].dayOfWeek" to "FRIDAY",
            "data[1].testInt" to "2"))
    assertThat(actual).isEqualTo(
        ObjectsWithOptionalEnumParameterHolder(
            listOf(
                InnerWithOptionalEnum(1, DayOfWeek.FRIDAY),
                InnerWithOptionalEnum(2, null))))
}
```

**Comment by abogushevsky** [comment]
Resolved after merging #21
